services/validatorWorker/lib/index.js

```diff
@@ -62,4 +62,48 @@ function toBNStringMap(raw){
 	return balances
 }
 
-module.exports = { getStateRootHash, isValidRootHash, toBNMap, getBalancesAfterFeesTree, toBNStringMap }
+function toStringBN(raw) {
+	let result = ``
+	Object.entries(raw).forEach(([acc, bal]) => result = `${result} ${acc}: ${bal.toString()}`)
+	return result
+}
+
+function invalidNewState(channel, adapter, { reason, newMsg}) {
+	// quirk: type is overiding type in newMsg
+	return persist(adapter, channel, {
+		...newMsg,
+		type: 'InvalidNewState',
+		reason,
+	})
+}
+
+function onError(channel, adapter, { reason, newMsg, prevBalances, newBalances }) {
```

Comment:
I'm pretty sure you don't need the balances here. For testing purposes, we only need to detect if the error type is the right one